src/marvin/components/ai_classifier.py

```diff
@@ -22,58 +32,168 @@ class ClassifierUser(User):
+            )
+        else:
+            # Call the parent class's __call__ method to create the enum
+            enum = super().__call__(
+                value,
+                names,
+                *values,
+                module=module,
+                qualname=qualname,
+                type=type,
+                start=start,
+                boundary=boundary,
+            )
+
+            # Set additional attributes for the AI classifier
+            setattr(enum, "__system__", system)
+            setattr(enum, "__user__", user)
+            setattr(enum, "__model__", model)
+            setattr(enum, "__value_getter__", value_getter)
+            return enum
+
+
+class AIEnum(Enum, metaclass=AIEnumMeta):
+    """
+    AIEnum is a class that extends Python's built-in Enum class.
+    It uses the AIEnumMeta metaclass, which allows additional parameters to be passed
+    when creating an enum. These parameters are used to customize the behavior
+    of the AI classifier.
+    """
+
+    @classmethod
+    def __messages__(
+        cls,
+        value,
+        system: System = ClassifierSystem,
+        user: User = ClassifierUser,
+        value_getter: Callable = lambda x: x.name,
+        as_dict: bool = True,
+        **kwargs,
+    ):
+        """
+        Generate the messages to be used as prompts for the AI classifier. The messages
+        are created based on the system and user templates provided.
+        """
+
+        messages = render_prompts(
+            [
+                system(
+                    enum_class_docstring=cls.__doc__ or "",
+                    options=[value_getter(option) for option in cls],
+                ),
+                user(user_input=value),
+            ],
+            render_kwargs=kwargs or {},
         )
-    if system_prompt is None:
-        system_prompt = ClassifierSystem
-    if user_prompt is None:
-        user_prompt = ClassifierUser
-    if value_getter is None:
 
-        def value_getter(x):
-            return x.name
+        if as_dict:
+            return [
+                {"role": message.role.value.lower(), "content": message.content}
+                for message in messages
+            ]
+        return messages
 
-    def decorator(enum_class):
-        # Add a new __missing__ method to enum_class to intercept NLP prompts
-        def _missing_(name):
-            messages = render_prompts(
-                [
-                    system_prompt(
-                        enum_class_docstring=enum_class.__doc__ or "",
-                        options=[value_getter(option) for option in enum_class],
-                    ),
-                    user_prompt(user_input=name),
-                ]
+    @classmethod
+    def _missing_(
+        cls,
+        value,
+        system: System = None,
+        user: User = None,
+        value_getter: Callable = None,
+        model: ChatLLM = None,
+        **kwargs,
```

Comment:
minor request to add `instructions` as an explicit optional kwarg here to document its existence as a valid thing to pass, otherwise it only shows up in the system prompt with no clear way to see how that gets populated (except tracing kwargs)